pkg/onit/cluster/gui.go

```diff
@@ -0,0 +1,132 @@
+      admin:
+        access_log_path: /tmp/admin_access.log
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 9901 }
+      static_resources:
+        listeners:
+        {{- range $key, $value := .Values.onosservices }}
+          - name: listener_{{$key}}
+            address:
+              socket_address: { address: 0.0.0.0, port_value: {{ $value.proxy }} }
+            filter_chains:
+              - filters:
+                  - name: envoy.http_connection_manager
+                    config:
+                      codec_type: auto
+                      stat_prefix: ingress_http
+                      route_config:
+                        name: local_route
+                        virtual_hosts:
+                          - name: local_service
+                            domains: ["*"]
+                            routes:
+                              - match: { prefix: "/" }
+                                route:
+                                  cluster: onos_{{ $key }}_service
+                                  max_grpc_timeout: 0s
+                            cors:
+                              allow_origin:
+                                - "*"
+                              allow_methods: GET, PUT, DELETE, POST, OPTIONS
+                              allow_headers: keep-alive,user-agent,cache-control,content-type,content-transfer-encoding,custom-header-1,x-accept-content-transfer-encoding,x-accept-response-streaming,x-user-agent,x-grpc-web,grpc-timeout
+                              max_age: "1728000"
+                              expose_headers: custom-header-1,grpc-status,grpc-message
+                      http_filters:
+                        - name: envoy.grpc_web
+                        - name: envoy.cors
+                        - name: envoy.router
+        {{- end }}
+        clusters:
+        {{- range $key, $value := .Values.onosservices }}
+          - name: onos_{{ $key }}_service
+            connect_timeout: 0.25s
+            type: logical_dns
+            http2_protocol_options: {}
+            lb_policy: round_robin
+            # win/mac hosts: Use address: host.docker.internal instead of address: localhost in the line below
+            hosts: [{ socket_address: { address: onos-{{$key}}, port_value: {{ $value.grpc }} }}]
+            tls_context:
+              common_tls_context:
+                tls_certificates:
+                  certificate_chain: { "filename": "/etc/envoy-proxy/certs/tls.crt" }
+                  private_key: { "filename": "/etc/envoy-proxy/certs/tls.key" }
+                validation_context:
+                  trusted_ca: { filename: "/etc/envoy-proxy/certs/tls.cacrt" }
+        {{- end }}`
+
+var envoyConfigMaps = map[string]string{
+	"/config/envoy.yaml": envoyConfig,
+}
+
+const (
+	guiType    = "gui"
+	guiImage   = "onosproject/onos-gui:latest"
+	envoyImage = "envoyproxy/envoy:v1.11.1"
+	guiService = "onos-gui"
+	guiPort    = 80
+)
```

Comment:
I think you should copy the values from 
https://github.com/onosproject/onos-helm-charts/blob/94701805c3cab7acc002fb7975c77d601c66bee5/onos-gui/values.yaml#L50
onos-services in here too
Envoy consumes 

- onos-config:5150 and exposes it on 8080
- onos-topo:5150 and exposes it on 8081

Then Nginx (in onos-gui container) maps

- http://onos-gui:80/onos-topo to http://localhost:8080
- http://onos-gui:80/onos-config to http://localhost:8081

(where localhost is the envoy sidecar container here)

I want 80, 8080 and 8081 all exposed through the service and available as endpoints

To run the GUI only reuires 80 to be exposed, but exposing 8080 and 8081 are needed for GUI development mode. 

The ingress only needs to deal with exposing 80